### PR TITLE
Cleaned up form-controls

### PR DESCRIPTION
**Pull request by samuelgrant**
Cleaned up form controls across the site to make them more consistent. 

**Code Review Checklist**
- [x] Ensure all forms still work: 
*Clear your cache before testing this*
   - Login
   - Password Reset
   - Request Password Reset
   - Register
   - Account (A users settings page)
   - Users (Manage all users page)

- [x] Code Quality
   - Code is clean
   - No potential issues found with visual inspection
   - Changes are consistent with the rest of this site

- [x] Special Checks
*Requested as part of this code cleanup*
   - All forms now use the component `<FormGroup>` to provide a label rather than the `label` element
   - Input attributes no longer need to have a value. `disabled="true"` should now simply be `disabled` (This is only when we call an **Input** or **Select** form control)
   - When calling a form component (Input or Select) any multi word props should be cammal cased. This means `autocomplete="email"` should now become `autoComplete="email"`

### CODE REVIEW

**Comment by samuelgrant** [comment]
@jacksos101 @MitchellQ Now ready for review & merge

File: Resources/js/React/Components/FormControl.js [hunk 1]
```diff
@@ -94,12 +87,22 @@ export class Input extends Component {
...
+                name={ this.props.name || false }
+                value={ this.state.value }
+                placeholder={ this.props.placeHolder || false }
+                autocomplete={ this.getAutoComplete() }
+                disabled={ this.props.disabled ? true : false }
+                readonly={ this.props.readOnly ? true : false }
+                required={ this.props.required ? true : false }
+                autoFocus={ this.props.autoFocus ? true: false }
+                minLength={ this.props.minLength || false }
+                maxLength={ this.props.minLength || false }
```
Comment (MitchellQ) [inline Resources/js/React/Components/FormControl.js]: max length should be set to `this.props.maxLength`

File: Resources/js/React/Authentication/Login.js [hunk 2]
```diff
@@ -54,19 +55,19 @@ export default class Login extends Component {
                     <h1 className="sr-only">Login Form</h1>
                     <h1 className="display-4 mb-5">Log In</h1>
 
-                    <div className="form-group">
-                        <input type="email" className="form-control" name="email" placeholder="Email" autoComplete="email" required/>
-                    </div>
+                    <FormGroup>
+                        <Input type="email" name="email" placeHolder="Email" autoComplete="email" required="true" autoFocus required/>
```
Comment (MitchellQ) [inline Resources/js/React/Authentication/Login.js]: Redundant required="true" needs to be removed

File: Resources/js/React/Components/FormControl.js [hunk 3]
```diff
@@ -94,12 +87,22 @@ export class Input extends Component {
...
-            <input id={id} type={this.getType()} className={this.getClass()} name={name} value={this.state.value} placeholder={placeholder} autocomplete={this.getAutoComplete()} disabled={this.isDisabled()} readonly={this.isReadOnly()} required={this.isRequired()} onChange={this.handleChange.bind(this)} />
+            <input id={ this.props.id || false }
+                type={this.getType()}
+                className={ this.props.inputClass || "form-control" }
+                name={ this.props.name || false }
+                value={ this.state.value }
+                placeholder={ this.props.placeHolder || false }
+                autocomplete={ this.getAutoComplete() }
+                disabled={ this.props.disabled ? true : false }
+                readonly={ this.props.readOnly ? true : false }
```
[the remark this answers is not in the document]
Comment (samuelgrant) [reply Resources/js/React/Components/FormControl.js]: I think so, I'll do that now. Honestly I always get confused between `!`, `!!` & `!!!` in JavaScript

[the remark this answers is not in the document]
Comment (samuelgrant) [reply Resources/js/React/Components/FormControl.js]: @jacksos101 Code push, #2b8b802 - Please review and test the double bangs

**Pull request merged**